phoenix-core/src/main/java/org/apache/phoenix/schema/task/Task.java

```diff
@@ -141,12 +216,68 @@ public static void addTask(PhoenixConnection conn, PTable.TaskType taskType, Str
                     PhoenixDatabaseMetaData.TASK_END_TS + ", " +
                     PhoenixDatabaseMetaData.TASK_DATA +
                     " ) VALUES(?,?,?,?,?,?,?,?,?)");
-            stmt = setValuesToAddTaskPS(stmt, taskType, tenantId, schemaName, tableName, taskStatus, data, priority, startTs, endTs);
-            LOGGER.info("Adding task " + taskType + "," +tableName + "," + taskStatus + "," + startTs, ","+endTs);
+            stmt = setValuesToAddTaskPS(stmt, systemTaskParams.getTaskType(),
+                systemTaskParams.getTenantId(),
+                systemTaskParams.getSchemaName(),
+                systemTaskParams.getTableName(),
+                systemTaskParams.getTaskStatus(), systemTaskParams.getData(),
+                systemTaskParams.getPriority(), systemTaskParams.getStartTs(),
+                systemTaskParams.getEndTs());
+            LOGGER.info("Adding task type: {} , tableName: {} , taskStatus: {}"
+                + " , startTs: {} , endTs: {}", systemTaskParams.getTaskType(),
+                systemTaskParams.getTableName(),
+                systemTaskParams.getTaskStatus(), systemTaskParams.getStartTs(),
+                systemTaskParams.getEndTs());
         } catch (SQLException e) {
             throw new IOException(e);
         }
-        mutateSystemTaskTable(conn, stmt, accessCheckEnabled);
+        // if query is getting executed by client, do not execute and commit
+        // mutations
+        if (isCommitAllowed) {
```

Comment:
Can we rename this variable? Sounds like it is set to `false` when we don't really want to commit, but rather just get the stmt object which we later use to get the `List<Mutation>`. This is used just for the index rebuild task that gets triggered from the client-side, right? For the tasks that are triggered from the server (i.e. from `MetaDataEndpointImpl` and `TaskRO`, we can just directly call `Task.addTask()` right?
A more appropriate name would be `shouldCommit` or something.

Reply:
> This is used just for the index rebuild task that gets triggered from the client-side, right? For the tasks that are triggered from the server (i.e. from `MetaDataEndpointImpl` and `TaskRO`, we can just directly call `Task.addTask()` right?

That's right, both of above are correct. Sure, let me rename it to `shouldCommit`, that one sounds better.